contracts/ChipRegistry.sol

```diff
@@ -137,225 +124,224 @@ contract ChipRegistry is IChipRegistry, ClaimedPBT, Ownable {
-        uint256 lockinPeriod = projectInfo.creationTimestamp + maxLockinPeriod > _chipClaim.developerMerkleInfo.lockinPeriod ?
-            _chipClaim.developerMerkleInfo.lockinPeriod :
+        uint256 lockinPeriod = projectInfo.creationTimestamp + maxLockinPeriod > projectInfo.lockinPeriod ?
+            projectInfo.lockinPeriod :
             projectInfo.creationTimestamp + maxLockinPeriod;
         
         // Set primaryService on ServicesRegistry
-        servicesRegistry.setInitialService(
+        _servicesRegistry.setInitialService(
             _chipId,
-            _chipClaim.developerMerkleInfo.serviceId,
+            projectInfo.serviceId,
             lockinPeriod
         );
 
-        ChipInfo memory chipInfo = ChipInfo({
-            tokenId: 0,     // temporary value, will be set in _mint
-            transferPolicy: projectInfo.transferPolicy,
-            tokenUri: _chipClaim.developerMerkleInfo.tokenUri,
-            tokenData: _encodeTokenData(_chipClaim.ersNode, _manufacturerValidation.enrollmentId)
-        });
-        // Mint a PBT this function fills out the ownership mapping, maps tokenId to chipId, fills out
-        // the chip table and increments the tokenIdCounter
-        uint256 tokenId = ClaimedPBT._mint(_chipClaim.owner, _chipId, chipInfo);
-
-        if (!projectInfo.claimsStarted) {
-            projectEnrollments[IProjectRegistrar(msg.sender)].claimsStarted = true;
+        if (!projectInfo.chipsAdded) {
+            projectEnrollments[projectRegistrar].chipsAdded = true;
         }
 
-        emit ChipClaimed(
+        emit ChipAdded(
             _chipId,
-            tokenId,
-            _chipClaim.owner,
-            _chipClaim.developerMerkleInfo.serviceId,
-            _chipClaim.ersNode,
+            address(projectRegistrar),
             _manufacturerValidation.enrollmentId,
-            _chipClaim.developerMerkleInfo.tokenUri
+            _chipOwner,
+            projectInfo.serviceId,
+            ersNode
         );
     }
 
+
     /**
-     * @notice Included for compliance with EIP-5791 standard but left unimplemented to ensure transfer policies can't be ignored.
+     * @dev ONLY Developer REGISTRAR: Remove project enrollment from ChipRegistry. This function is only callable by DeveloperRegistrars. This
+     * function will revert if the project is not enrolled or if the project has already added chips. This function will also remove the project
+     * subnode record in the ERS.
+     *
+     * @param _projectRegistrar          Address of the ProjectRegistrar contract
      */
-    function transferTokenWithChip(
-        bytes calldata /*signatureFromChip*/,
-        uint256 /*blockNumberUsedInSig*/,
-        bool /*useSafeTransfer*/
-    )
-        public
-        virtual
-        override(ClaimedPBT, IPBT)
-    {
-        revert("Not implemented");
+    function removeProjectEnrollment(IProjectRegistrar _projectRegistrar) external {
```

Comment:
We need a check that the calling `DeveloperRegistrar` is the owner of the project. Either need to do that here or do that on the `DeveloperRegistrar`. In this instance it can be done on the `DeveloperRegistrar` because it is a trusted contract

Reply:
Missed that it's on Line 291 but not sure that check is sufficient since the `ProjectRegistrar` is not a trusted contract

Reply:
Added in WIP commit.